vendor/bat-native-ads/src/bat/ads/internal/account/ad_rewards/ad_rewards.cc

```diff
@@ -208,24 +208,13 @@ base::Value AdRewards::GetAsDictionary() {
 bool AdRewards::SetFromDictionary(base::Value* dictionary) {
   DCHECK(dictionary);
 
-  base::Value* ad_rewards = dictionary->FindDictKey("ads_rewards");
-  if (!ad_rewards) {
-    return false;
-  }
-
-  base::DictionaryValue* ads_rewards_dictionary;
-  if (!ad_rewards->GetAsDictionary(&ads_rewards_dictionary)) {
-    return false;
-  }
```

Comment:
The bug was due to the passed-in `dictionary` already containing the `ads_rewards` key/values